run_brer/run_config.py

```diff
@@ -583,6 +583,8 @@ def run(self, tpr_file=None, **kwargs):
             else:
                 assert end_time > start_time
                 if end_time - start_time >= requested_production_time:
+                    prev_iter_state_json = 'state_' + str(self.run_data.get('iteration')) + '.json'
+                    shutil.copy2(self.state_json, prev_iter_state_json)
```

Comment:
How should it behave if the file already exists?

Reply:
If `prev_iter_state_json` already exists? (i.e. we're now on iteration 2 or 3). I was assuming that the different iteration values would result in an overwriting of `prev_iter_state_json` as a variable name but not the content.

Reply:
Well, in the simple case, we expect `state_<iteration>.json` to be a new file. What if it isn't? Is it always an error if the file already exists or is there a legitimate reason why we might reach this point twice? If it is an error for the file to already exist, how should we handle/report the error? Is it a minor enough error that we should just overwrite it?

Note that `shutil.copy2()` will usually quietly replace a destination file if it already exists, so if we want to prevent that, we need to check for ourselves. https://docs.python.org/3/library/shutil.html#shutil.copy

Reply:
I don't think we should over-engineer a solution to a problem we don't understand well. My proposal would probably be to
1. Check whether the destination file already exists.
2. If it exists, error out and tell the user that if they intended to re-run the phase, they should remove the file first.
(The message should give the full path.)

If we start to encounter the situation and we aren't happy with this approach, we can consider other options.

Reply:
If you think it is better to issue only a warning and proceed, I'm fine with that, too.

Reply:
Added proposed changes to my commit!